src/utils/event-manager/wheel-input.js

```diff
@@ -0,0 +1,126 @@
+/* global window:false */
+const ua = typeof window.navigator !== 'undefined' ?
+  window.navigator.userAgent.toLowerCase() : '';
+const firefox = ua.indexOf('firefox') !== -1;
+
+const WHEEL_EVENTS = [
+  // Chrome, Safari
+  'wheel',
+  // IE
+  'mousewheel',
+  // legacy Firefox
+  'DOMMouseScroll'
+];
+
+export default class WheelInput {
+
+  constructor(element, callback) {
+    this.element = element;
+    this.callback = callback;
+
+    this.handler = this.handler.bind(this);
+
+    WHEEL_EVENTS.forEach(eventName => element.addEventListener(eventName, this.handler));
+
+    this._state = {
+      mouseWheelPos: null
+    };
+  }
+
+  destroy() {
+    const {element} = this;
+    WHEEL_EVENTS.forEach(eventName => element.removeEventListener(eventName, this.handler));
+  }
+
+  /* eslint-disable complexity, max-statements */
+  handler(event) {
+
+    event.preventDefault();
+    let value = event.deltaY;
+    // Firefox doubles the values on retina screens...
+    if (firefox && event.deltaMode === window.WheelEvent.DOM_DELTA_PIXEL) {
+      value /= window.devicePixelRatio;
+    }
+    if (event.deltaMode === window.WheelEvent.DOM_DELTA_LINE) {
+      value *= 40;
```

Comment:
a few of these numbers are used again, e.g. `40` and `4` - can we extract them to constants at the top? it'll also give more context to why these specific values were chosen